libs/telemetry/src/main/java/org/opensearch/telemetry/tracing/SpanCreationContext.java

```diff
@@ -11,24 +11,88 @@
 import org.opensearch.common.annotation.ExperimentalApi;
 import org.opensearch.telemetry.tracing.attributes.Attributes;
 
+import java.util.Objects;
+
 /**
  * Context for span details.
  *
  * @opensearch.experimental
  */
 @ExperimentalApi
 public final class SpanCreationContext {
-    private final String spanName;
-    private final Attributes attributes;
+    private String spanName;
+    private Attributes attributes;
+    private SpanKind spanKind = SpanKind.INTERNAL;
+    private SpanContext parent;
+
+    /**
+     * Factory method to create {@link SpanCreationContext}
+     * @return spanCreationContext
+     */
+    public static SpanCreationContext create() {
+        return new SpanCreationContext();
+    }
 
     /**
      * Constructor.
+     */
+    private SpanCreationContext() {}
+
+    /**
+     * Sets the span type to server.
+     * @return spanCreationContext
+     */
+    public SpanCreationContext server() {
```

Comment:
This should be `static` methods (replacing `create`), otherwise the API is backward: `SpanCreationContext.create().client().server().name().internal()`